### PR TITLE
[3fd|python3] switch to vcpkg-msbuild

### PR DESCRIPTION
**Pull request by Neumann-A**
- make 3fd use vcpkg-msbuild
- make python use vcpkg-msbuild
- fix python cross builds on windows

### CODE REVIEW

File: ports/python3/portfile.cmake [hunk 1]
```diff
@@ -94,11 +95,15 @@ if(VCPKG_TARGET_IS_WINDOWS)
         find_library(SQLITE_DEBUG NAMES sqlite3 PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
         find_library(SSL_RELEASE NAMES libssl PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH)
         find_library(SSL_DEBUG NAMES libssl PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
+        list(APPEND add_libs_rel "${BZ2_RELEASE};${EXPAT_RELEASE};${FFI_RELEASE};${LZMA_RELEASE};${SQLITE_RELEASE}")
+        list(APPEND add_libs_dbg "${BZ2_DEBUG};${EXPAT_DEBUG};${FFI_DEBUG};${LZMA_DEBUG};${SQLITE_DEBUG}")
```
Comment (dg0yt) [inline ports/python3/portfile.cmake:99]: Wouldn't this benefit from `x_vcpkg_pkgconfig_get_modules()`?

Comment (Neumann-A) [reply ports/python3/portfile.cmake:99]: I could rather add `DEPENDENT_PKGCONFIG` to `vcpkg_msbuild_install` but didn't want to go there yet.

Comment (Osyotr) [reply ports/python3/portfile.cmake:99]: Why are these libs duplicated in props file?

Comment (Neumann-A) [reply ports/python3/portfile.cmake:99]: The ultimate cleanup plan would be to remove `openssl.props.in` and `python_vcpkg.props.in` completely.

**Comment by Neumann-A** [comment]
> CI run failed:

Yeah I know. Something wrong with msbuild not selecting the right configuration for some reason. The command line has `/p:Configuration=Release` and `/p:Platform=Win32` so I don't quite know why it is selecting `x64` instead

**Comment by Neumann-A** [comment]
hmm even `E:\vcpkg\buildtrees\python3\x86-windows-rel>msbuild E:/vcpkg/buildtrees/python3/x86-windows-rel/PCbuild/pcbuild.proj -p:Platform=Win32` doesn't call the correct compiler from `vcpkg env --triplet x86-windows`

**Comment by Neumann-A** [comment]
Hmm there is a bug in python3 which builds an executable for the host instead the target which is why the cross builds fail here (see https://github.com/microsoft/vcpkg/pull/31727#issuecomment-1669460897)

File: ports/python3/portfile.cmake (hunk 1 above)
Comment (Hoikas) [reply ports/python3/portfile.cmake:102]: Why?

File: ports/python3/python_vcpkg.props.in [hunk 2]
```diff
@@ -8,11 +8,6 @@
       <AdditionalIncludeDirectories>
         %(AdditionalIncludeDirectories);${CURRENT_INSTALLED_DIR}/include
       </AdditionalIncludeDirectories>
-
-      <RuntimeLibrary Condition="'${VCPKG_CRT_LINKAGE}|$(Configuration)' == 'static|Debug'">MultiThreadedDebug</RuntimeLibrary>
-      <RuntimeLibrary Condition="'${VCPKG_CRT_LINKAGE}|$(Configuration)' == 'static|Release'">MultiThreaded</RuntimeLibrary>
-      <RuntimeLibrary Condition="'${VCPKG_CRT_LINKAGE}|$(Configuration)' == 'dynamic|Debug'">MultiThreadedDebugDLL</RuntimeLibrary>
-      <RuntimeLibrary Condition="'${VCPKG_CRT_LINKAGE}|$(Configuration)' == 'dynamic|Release'">MultiThreadedDLL</RuntimeLibrary>
```
Comment (Hoikas) [inline ports/python3/python_vcpkg.props.in:15]: Have you tested that all of these configurations work as expected? I don't remember what exactly this fixed, but I know this was in here for a reason.

File: ports/3fd/portfile.cmake [hunk 3]
```diff
@@ -13,18 +13,16 @@ vcpkg_from_github(
 )
 
 # Build:
-if (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore") # UWP:
+if (VCPKG_TARGET_IS_UWP) # UWP:
     vcpkg_install_msbuild(
```
Comment (dg0yt) [inline ports/3fd/portfile.cmake:17]: Is this to use `vcpkg_msbuild_install`, too?

File: ports/python3/portfile.cmake (hunk 1 above)
Comment (Neumann-A) [reply ports/python3/portfile.cmake:102]: @Hoikas I assume the why is in response of dropping the `props` files as ultimate cleanup. Simply said no portfile should require a direct interaction with build system internals. In the case of MSBuild it is the injection of props files which every port would have needed to to manually. It is simply better to have an abstraction around that than having to deal with msbuild shenanigans.

File: ports/python3/python_vcpkg.props.in (hunk 2 above)
Comment (Neumann-A) [reply ports/python3/python_vcpkg.props.in:15]: a) CI tests the correctness by post build checks.
b) Those values are not gone but integrated in the props/target file generated by `vcpkg_msbuild_install`

File: ports/3fd/portfile.cmake (hunk 3 above)
Comment (Neumann-A) [reply ports/3fd/portfile.cmake]: Hmm would have assumed CI to catch that but I see `3fd:x64-uwp:  cascade`. So the port once build for uwp but no longer does.

**Review by Adela0814 — APPROVED** [review]

**Comment by Osyotr** [comment]
So the official python installation does not contain `_freeze_module.exe`. This tool is used only at build time thus can be removed from the installed tree. Is this bad that it's built with real host arch?

**Comment by Neumann-A** [comment]
> Is it bad that it's built with real host arch?

Yes since that is against vcpkg policy. vcpkg only builds for the target triplet nothing else.
 
 > This tool is used only at build time thus can be removed from the installed tree
 
 No it cannot since it is required to run for cross builds to generate some headers/scripts if I understand what it is doing correctly. I moved it to `manual-tools/python3` so that it won't get picked up by CMake or whatever else might go looking for .exe files.

**Pull request merged**